R/realtime.service.R

```diff
@@ -0,0 +1,182 @@
+                         "WaveWatch3 model 7-day forecast, were used to compute wave setup at the shoreline.")
+      tiny.text <- paste("Modeled", gsub("mean", "Mean", gsub("extreme", "Probability", tiny.text)))
+    }
+    
+    full.publications  <-	getPublications(doc)
+    keywords	<-	getKeywords (doc,subType)
+  }
+	
+	summaryJSON	<- toJSON(list(
+		'version'=as.character(packageVersion(getPackageName())),
+		'tiny'=list('text'=tiny.text),
+		'medium'=list('title'=medium.title,'text'=medium.text),
+		'full'=list('title'=full.title,'text'=full.text,'publications'=full.publications),
+		'keywords'=keywords), auto_unbox = TRUE )
+	summaryJSON	<- sub('NULL. ','',summaryJSON)
+	return(summaryJSON)
+}
+
+extractInfoFromTitle = function(title) {
+  # Hurricane Sandy Assessment of Potential Coastal-Change Impacts: NHC Advisory 29, 1100 AM EDT MON OCT 29 2012
+  match <- regexec("(\\w+ \\w+).*(NHC Advisory (\\d+), (.*))", title)
+  matches <- regmatches(title, match)
+  if (length(matches)) {
+    parts <- as.list(matches[[1]][c(2, 3, 4, 5)])
+  } else {
+    parts <- as.list(rep("",4))
+  }
+  names(parts)<- c("name", "advFull", "advNum", "time")
+  return(parts)
+}
+
+extractCollectionDate = function(doc, attr) {
+  attrDescs <- sapply(getNodeSet(doc,paste0('//eainfo/detailed/attr')), xmlChildren)
+  matchingAttr <- Filter(function(f){
+    grepl(attr, xmlValue(f$attrlabl), ignore.case = TRUE)
+  }, attrDescs)
+  defn <- xmlValue(matchingAttr[[1]]$attrdef)
+  match <- regexec(".*?(\\w+ \\d{4} to \\w+ \\d{4})\\.?", defn)
+  collected <- regmatches(defn, match)[[1]][2]
+  return(collected)
+}
+
+getSurgeDescription = function(doc) {
+  text <- paste("The storm surge elevations along the open coast were obtained from the",
+                "National Oceanic and Atmospheric Administration's (NOAA)")
+  srcUsed <- sapply(getNodeSet(doc, '//dataqual/lineage/procstep/srcused'), xmlValue)
+  if (has(srcUsed, "psurge", ignore.case=TRUE)) {
+    text <- paste(text, "probabilistic surge forecast (psurge), which is based on conditions specific to the",
+                       "landfalling storm. Errors in hurricane forecasts are included in order to identify probable surge levels.",
+                       "The 10% exceedance surge level was used to represent the worst-case scenario.")
+  } else if (has(srcUsed, "estofs", ignore.case=TRUE)) {
+    text <- paste(text, "ESTOFS (Extratropical Surge and Tide Operational Forecast System).")
+  } else if (has(srcUsed, "moms", ignore.case=TRUE)) {
+    text <- paste(text, "Sea, Lake, and Overland Surges from Hurricanes (SLOSH) model, maximum of the maximum (MOM).")
+  } else {
+    # this shouldn't happen
+    text <- paste(text, ".")
+  }
+  return(text)
+}
+
+getPCOIDescription = function(subType, titleParts, ...) {
+  definition <- paste("probability of",
+                  paste0(titleMap$full[[subType]], ","), "or the likelihood",
+                  "that wave runup and storm surge will",
```

Comment:
per Heather's comments, pind medium description should be slightly different:

... or the likelihood that storm surge and wave setup will ...

rather than what you have here on line 160